presto-main/src/test/java/com/facebook/presto/server/TestQueryProgressStats.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import io.airlift.json.JsonCodec;
+import org.joda.time.DateTime;
+import org.testng.annotations.Test;
+
+import java.util.OptionalDouble;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestQueryProgressStats
+{
+    @Test
+    public void testJson()
+    {
+        QueryProgressStats expected = new QueryProgressStats(
+                DateTime.parse("1991-09-06T05:00-05:30"),
+                123456,
+                1111,
+                22222,
+                3333,
+                444,
+                34230492,
+                1000,
+                100000,
+                false,
+                OptionalDouble.of(33.33));
+        JsonCodec<QueryProgressStats> codec = JsonCodec.jsonCodec(QueryProgressStats.class);
+
+        String json = codec.toJson(expected);
+        QueryProgressStats actual = codec.fromJson(json);
+
+        assertEquals(actual.getExecutionStartTime().getMillis(), expected.getExecutionStartTime().getMillis());
```

Comment:
Note that we actually made a mistake with constructor arguments in one of the query events which caused us to lose a bunch of logging data.